### PR TITLE
refactor!: use Google Meet API

### PR DESCRIPTION
**Pull request by alexandre-abrioux**
### Breaking Change
Calendar events are not created anymore. Google Meets won't be assigned to Calendar events.

### Changes
- Use the [new Google Meet API](http://workspaceupdates.googleblog.com/2024/02/google-meet-api-now-generally-available.html) instead of the Calendar API (no need for a `CALENDAR_ID` anymore)
- Replace yarn by pnpm
- Add script to generate the OAuth refresh token
- Update doc
- Update dependencies
- Update Node.js to Current (v22)

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -2,20 +2,26 @@
...
+    "@google-cloud/local-auth": "^3.0.1",
+    "dotenv": "^16.4.5",
+    "express": "^4.19.2"
   },
-  "license": "MIT",
   "devDependencies": {
-    "backpack-core": "^0.8.4",
-    "@types/node": "^15.3.1"
+    "prettier": "^3.3.0",
+    "google-auth-library": "^9.10.0"
```
Comment (benjlevesque) [inline package.json]: why is this a dev dependency? it seems to be used in `meet.js`. Did you mean to set `@google-cloud/local-auth`  here?

**Review by benjlevesque — APPROVED** [review]
This is great, I didn't know about this new API, thanks a lot @alexandre-abrioux!

few (sort of unrelated) notes, absolutely not blocking:
- slack-meet could easily be ran as a cloud function, maybe the repo should make it simpler
- the refresh_token seems kind of redundant with the client ID and Secret... Did you dig why it is necessary? I remember struggling with this last time I worked on that


FYI, I've made you a collaborator :)

**Comment by benjlevesque** [comment]
ignore my comment about refresh token, it's totally required because you need to impersonate someone (the dev user who ran the script)
tried with a SA but doesn't work...

File: package.json (hunk 1 above)
Comment (alexandre-abrioux) [reply package.json]: Ha yes I got it backwards! Fixed in https://github.com/benjlevesque/slack-meet/pull/175/commits/1eef6b92ca925cb8fcbe6e7c481e2ea5024ee888
Thanks!

**Comment by alexandre-abrioux** [comment]
> ignore my comment about refresh token, it's totally required because you need to impersonate someone (the dev user who ran the script) tried with a SA but doesn't work...

Yes, exactly, I tried with a service account, too, but it can only be done with a personal token 🤒

> slack-meet could easily be ran as a cloud function, maybe the repo should make it simpler

Yes, or we could even [have a button to deploy it directly to Cloud Run](https://cloud.google.com/blog/products/serverless/introducing-cloud-run-button-click-to-deploy-your-git-repos-to-google-cloud) + a CI to auto-build and push the new Docker image on release.

**Comment by alexandre-abrioux** [comment]
@benjlevesque I will work on a CI job to build and push the Docker image automatically. In the meantime, feel free to merge this and manually release the new Docker image whenever 🙂

**Pull request merged**